core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -5538,7 +5538,7 @@ ImmutableList<RelNode> retrieveCursors() {
         case ARRAY_QUERY_CONSTRUCTOR:
           call = (SqlCall) expr;
           query = Iterables.getOnlyElement(call.getOperandList());
-          root = convertQueryRecursive(query, false, null);
+          root = convertQueryRecursive(query, true, null);
```

Comment:
I think this deserves a comment. I think you are pretending this to be the toplevel query, because this will prevent ORDER BY from being eliminated.

Reply:
Agree. Perhaps rename `boolean top` to `boolean preserveOrderBy` (assuming that `top` doesn't control any other behavior). Then the documentation can be on the `convertQueryRecursive` method.

Reply:
thanks, fixed. however I still used top, The current codepath is passed in from top all the way. top is also easy to understand.

Reply:
You force-pushed?!

Reply:
> You force-pushed?!

Sorry, I know the rules for review, but I accidentally force-push it ..